client/src/main/java/io/pravega/client/byteStream/impl/ByteStreamClientImpl.java

```diff
@@ -0,0 +1,62 @@
+/**
+ * Copyright (c) 2018 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.client.byteStream.impl;
+
+import io.pravega.client.byteStream.ByteStreamClient;
+import io.pravega.client.byteStream.ByteStreamReader;
+import io.pravega.client.byteStream.ByteStreamWriter;
+import io.pravega.client.segment.impl.Segment;
+import io.pravega.client.segment.impl.SegmentInputStreamFactory;
+import io.pravega.client.segment.impl.SegmentMetadataClientFactory;
+import io.pravega.client.segment.impl.SegmentOutputStreamFactory;
+import io.pravega.client.stream.EventWriterConfig;
+import io.pravega.client.stream.impl.Controller;
+import io.pravega.common.concurrent.Futures;
+import lombok.RequiredArgsConstructor;
+
+@RequiredArgsConstructor
+public class ByteStreamClientImpl implements ByteStreamClient {
+
+    private final String scope;
+    private final Controller controller;
+    private final SegmentInputStreamFactory inputStreamFactory;
+    private final SegmentOutputStreamFactory outputStreamFactory;
+    private final SegmentMetadataClientFactory metaStreamFactory;
+
+    @Override
+    public ByteStreamReader createByteStreamReader(String streamName) {
+        return createByteStreamReaders(new Segment(scope, streamName, 0));
+    }
+
+    ByteStreamReader createByteStreamReaders(Segment segment) {
+        String delegationToken = Futures.getAndHandleExceptions(controller.getOrRefreshDelegationTokenFor(segment.getScope(),
+                                                                                                          segment.getStream()
+                                                                                                                 .getStreamName()),
+                                                                RuntimeException::new);
+        return new ByteStreamReaderImpl(inputStreamFactory.createInputStreamForSegment(segment, delegationToken),
+                                        metaStreamFactory.createSegmentMetadataClient(segment, delegationToken));
+    }
+
+    @Override
+    public ByteStreamWriter createByteStreamWriter(String streamName) {
+        return createByteStreamWriter(new Segment(scope, streamName, 0), EventWriterConfig.builder().build());
```

Comment:
Where is this segment actually created? I'm wondering about the case in which multiple writers append (not simultaneously) to the same byte stream. Could you clarify the expected behavior, please?

Reply:
My expectation is they have created the stream via normal means using the StreamManager API. If they have not done so they will get an exception.

Reply:
I can create a stream with say 100 segments and if I use the byte stream writer API against it, only segment 0 will be used, is that right? I need to double-check that we describe this behavior somewhere.

Reply:
Yeah, perhaps I should add a check to make sure there is only one segment?
Related question: Should I add a flag here that's "throwIfNotEmpty"? After all most of the time using this API on a non-empty segment is undesireable.